go/obscuronode/enclave/l1_blockchain.go

```diff
@@ -0,0 +1,122 @@
+package enclave
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+
+	"github.com/ethereum/go-ethereum/consensus"
+	"github.com/ethereum/go-ethereum/consensus/beacon"
+	"github.com/ethereum/go-ethereum/consensus/ethash"
+	"github.com/ethereum/go-ethereum/core"
+	"github.com/ethereum/go-ethereum/core/rawdb"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/core/vm"
+	"github.com/ethereum/go-ethereum/eth/ethconfig"
+	"github.com/ethereum/go-ethereum/ethdb"
+	"github.com/ethereum/go-ethereum/params"
+)
+
+// NewL1Blockchain creates a Geth BlockChain object. `genesisJSON` is the Genesis block config in JSON format. A Geth
+// node can be made to output this using the `dumpgenesis` startup command.
+func NewL1Blockchain(genesisJSON []byte) *core.BlockChain {
+	dataDir, err := ioutil.TempDir(os.TempDir(), "")
+	if err != nil {
+		panic(err)
+	}
+
+	db := createDB(dataDir)
+	cacheConfig := createCacheConfig(dataDir)
+	chainConfig := createChainConfig(db, genesisJSON)
+	engine := createEngine(dataDir)
+	vmConfig := createVMConfig()
+	shouldPreserve := createShouldPreserve()
+	txLookupLimit := ethconfig.Defaults.TxLookupLimit // Default.
+
+	blockchain, err := core.NewBlockChain(db, cacheConfig, chainConfig, engine, vmConfig, shouldPreserve, &txLookupLimit)
+	if err != nil {
+		panic(fmt.Errorf("l1 blockchain could not be created: %w", err))
+	}
+	return blockchain
+}
+
+func createDB(dataDir string) ethdb.Database {
+	root := path.Join(dataDir, "geth/chaindata")            // Defaults to `geth/chaindata` in the node's data directory.
+	cache := 2048                                           // Default.
```

Comment:
pls raise a ticket to add all these to the config framework ( after we introduce it)

Reply:
Done.